working/macros/api/introspection.dart

```diff
@@ -39,6 +39,24 @@ abstract class NamedTypeAnnotation implements TypeAnnotation {
   Iterable<TypeAnnotation> get typeArguments;
 }
 
+/// The interface representing a resolved type.
+///
+/// Resolved types understand exactly what type they represent, and can be
+/// compared to other static types.
+abstract class StaticType {
+  /// Returns true if this is a subtype of [other].
+  bool isSubtypeOf(StaticType other);
```

Comment:
Is it right that `StaticType` related APIs are synchronous?

If macro builders run in a separate isolate, all the type and declaration information is in the main isolate, e.g. in form of the analyzer element model. So, almost everything is RPC.